wallets/provider-ledger/src/helpers.ts

```diff
@@ -44,6 +54,35 @@ export async function getLedgerAccounts(): Promise<{
   }
 }
 
+export async function getSolanaAccounts(): Promise<{
+  accounts: string[];
+  chainId: string;
+}> {
+  try {
+    const transport = await transportConnect();
+
+    const solana = new (await import('@ledgerhq/hw-app-solana')).default(
+      transport
+    );
+
+    const accounts: string[] = [];
+
+    const result = await solana.getAddress(SOLANA_BIP32_PATH);
+    if (result?.address) {
```

Comment:
In what situation `result` can be `undefined`? shouldn't we thrown error if `result` is undefined?

Reply:
removed the if statement